tests/rustdoc-ui/normalize-in-inlined-type-alias.rs

```diff
@@ -0,0 +1,14 @@
+// check-pass
+// compile-flags: -Znormalize-docs
+
+trait Woo<T> {
+    type Assoc;
+}
+
+impl<T> Woo<T> for () {
+    type Assoc = ();
+}
+
+type Alias<P> = <() as Woo<P>>::Assoc;
+
+pub fn hello<S>() -> Alias<S> {}
```

Comment:
Specifically, here we were trying to normalize `<() as Woo<P>>::Assoc` (the type of `Alias<P>`) in a param-env that was `[S: Sized]`.